lib/ui/profitloss/RecordAmountListItem.dart

```diff
@@ -0,0 +1,357 @@
+                height: style.detailTextSpacing)
+          ]));
+    }
+
+    return listBuilder;
+  }
+
+  List<Widget> _buildDescriptionTextField(List<Widget> listBuilder, RecordAmountListItemViewModel viewModel, RecordAmountListItemStyle style) {
+    listBuilder.add(Expanded(
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          viewModel.isEditable
+              ? TextField(
+                  decoration: InputDecoration(
+                      hintText: viewModel.hint,
+                      hintStyle: style.pendingDetailTextStyle,
+                      border: InputBorder.none,
+                      contentPadding: style.cardMargins,
+                      counterText: ""),
+                  textAlign: TextAlign.left,
+                  style: style.detailTextStyle,
+                  maxLines: style.maxLines,
+                  controller: _textFieldController,
+                  onEditingComplete: () => _checkTextField(viewModel),
+                  enabled:viewModel.isEditable,
+                )
+              : Text(viewModel.description,
+                  textAlign: TextAlign.left,
+                  style: style.detailTextStyle)
+        ],
+      ),
+    ));
+    return listBuilder;
+  }
+
+  void _buildItemPicker(List<Widget> listBuilder, RecordAmountListItemViewModel viewModel, RecordAmountListItemStyle style) {
+    listBuilder.add(
+        Text(viewModel.title, style: style.titleTextStyle));
+    listBuilder.add(Expanded(
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        crossAxisAlignment: CrossAxisAlignment.end,
+        children: <Widget>[
+          viewModel.isEditable
+              ? DropdownButtonHideUnderline(
+                  child: DropdownButton(
+                    value: viewModel.selectedItem,
+                    style: style.detailTextStyle,
+                    items: viewModel.isEditable
+                        ? _getDropDownMenuItems(viewModel)
+                        : null,
+                    onChanged: _changeDropDownItem,
+                    hint: Text(viewModel.hint,
+                        style: style.pendingDetailTextStyle),
+                    icon: Icon(Icons.lens, size: 0),
+                    isDense: true,
+                    isExpanded: false,
+                  ),
+                )
+              : Text(viewModel.selectedItem,
+                  style: style.detailTextStyle)
+        ],
+      ),
+    ));
+  }
+
+  void _buildDatePicker(List<Widget> listBuilder, RecordAmountListItemViewModel viewModel, RecordAmountListItemStyle style) {
+    listBuilder.add(
+        Text(viewModel.title, style: style.titleTextStyle));
+
+    listBuilder.add(Expanded(
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.end,
+        children: <Widget>[
+          InkWell(
+              child: Text(
+                  _formatDate(viewModel.selectedDate) ==
```

Comment:
maybe put this is a final above to be more clear

Reply:
Done